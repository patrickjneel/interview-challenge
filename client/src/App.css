* {
   font-family: 'Roboto Mono', monospace;
}

.app {
  align-items: center;
  background: url('./building.jpg');
  background-size: cover;
  display: flex;
  justify-content: center;
  height: 100vh; 
}

.top-app {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.title {
  color: #ffffff;
  font-size: 72px;
  font-weight: 700;
  margin: unset;
}

.desc {
  color: #efefef;
  font-weight: 300;
  /*margin: unset;*/
}


