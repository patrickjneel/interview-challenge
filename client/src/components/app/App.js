import React, { Component } from 'react';
import fetchData from '../api/API';
import ChartArea from '../chartArea/ChartArea';
import './app.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      data: []
    }
  }

  async componentDidMount() {
    const data = await fetchData();
    this.setState({ data })
  }

  scroll = () => {
    const chart = document.querySelector('.charts')
    chart.scrollIntoView({ behavior: "smooth", block: 'start' })
  }

  render() {
    return (
      <div className='app'>
        <div className='welcome-screen'>
          <div className='top-app'>
            <h1 className='title'>ADCELLERANT</h1>
            <h2 className='desc'>Digital Advertising & Technology Company</h2>
          </div>
          <div className='bottom-app'>
            <h3 className='data-desc' onClick={this.scroll}>View Mobile Data</h3>
            <img className='logo' src={require('../../assets/creative.svg')} />
            <h3 className='data-desc' onClick={this.scroll}>View Tablet Data</h3>
          </div>
        </div>
        <div className='charts'>
          <ChartArea data={this.state.data} />
        </div>
      </div>
    );
  };
};

export default App;
