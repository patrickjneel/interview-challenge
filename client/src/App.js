import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      data: []
    }
  }
  componentDidMount() {
    fetch('/data').then(data => data.json()).then(info => console.log(info.rows))
  }
  render() {
    return (
      <div className='app'>
        <div className='top-app'>
          <h1 className='title'>ADCELLERANT</h1>
          <h2 className='desc'>Digital Advertising & Technology Company</h2>
        </div>
      </div>
    );
  }
}

export default App;
