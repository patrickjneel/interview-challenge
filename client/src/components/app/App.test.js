import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App Test', () => {

  it('should be defined', () => {
    const mockData = []
    const renderedApp = shallow(<App data={mockData}/>);

    expect(renderedApp).toBeDefined();
  });

  it('should match the snap shot', () => {
    const renderedApp = shallow(<App />);

    expect(renderedApp).toMatchSnapshot();
  });
})
