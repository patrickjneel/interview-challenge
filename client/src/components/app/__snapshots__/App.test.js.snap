// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test should match the snap shot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="welcome-screen"
>
          <div
                    className="top-app"
          >
                    <h1
                              className="title"
                    >
                              ADCELLERANT
                    </h1>
                    <h2
                              className="desc"
                    >
                              Digital Advertising & Technology Company
                    </h2>
          </div>
          <div
                    className="bottom-app"
          >
                    <h3
                              className="data-desc"
                              onClick={[Function]}
                    >
                              View Mobile Data
                    </h3>
                    <img
                              className="logo"
                              src="creative.svg"
                    />
                    <h3
                              className="data-desc"
                              onClick={[Function]}
                    >
                              View Tablet Data
                    </h3>
          </div>
          <h3
                    className="data-desc"
                    onClick={[Function]}
          >
                    View Comparison Data
          </h3>
</div>,
        <div
          className="charts"
>
          <ChartArea
                    data={Array []}
          />
</div>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="top-app"
>
              <h1
                            className="title"
              >
                            ADCELLERANT
              </h1>
              <h2
                            className="desc"
              >
                            Digital Advertising & Technology Company
              </h2>
</div>,
            <div
              className="bottom-app"
>
              <h3
                            className="data-desc"
                            onClick={[Function]}
              >
                            View Mobile Data
              </h3>
              <img
                            className="logo"
                            src="creative.svg"
              />
              <h3
                            className="data-desc"
                            onClick={[Function]}
              >
                            View Tablet Data
              </h3>
</div>,
            <h3
              className="data-desc"
              onClick={[Function]}
>
              View Comparison Data
</h3>,
          ],
          "className": "welcome-screen",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="title"
>
                  ADCELLERANT
</h1>,
                <h2
                  className="desc"
>
                  Digital Advertising & Technology Company
</h2>,
              ],
              "className": "top-app",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ADCELLERANT",
                  "className": "title",
                },
                "ref": null,
                "rendered": "ADCELLERANT",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Digital Advertising & Technology Company",
                  "className": "desc",
                },
                "ref": null,
                "rendered": "Digital Advertising & Technology Company",
                "type": "h2",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="data-desc"
                  onClick={[Function]}
>
                  View Mobile Data
</h3>,
                <img
                  className="logo"
                  src="creative.svg"
/>,
                <h3
                  className="data-desc"
                  onClick={[Function]}
>
                  View Tablet Data
</h3>,
              ],
              "className": "bottom-app",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "View Mobile Data",
                  "className": "data-desc",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "View Mobile Data",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "logo",
                  "src": "creative.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "View Tablet Data",
                  "className": "data-desc",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "View Tablet Data",
                "type": "h3",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "View Comparison Data",
              "className": "data-desc",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "View Comparison Data",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ChartArea
            data={Array []}
/>,
          "className": "charts",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="welcome-screen"
>
            <div
                        className="top-app"
            >
                        <h1
                                    className="title"
                        >
                                    ADCELLERANT
                        </h1>
                        <h2
                                    className="desc"
                        >
                                    Digital Advertising & Technology Company
                        </h2>
            </div>
            <div
                        className="bottom-app"
            >
                        <h3
                                    className="data-desc"
                                    onClick={[Function]}
                        >
                                    View Mobile Data
                        </h3>
                        <img
                                    className="logo"
                                    src="creative.svg"
                        />
                        <h3
                                    className="data-desc"
                                    onClick={[Function]}
                        >
                                    View Tablet Data
                        </h3>
            </div>
            <h3
                        className="data-desc"
                        onClick={[Function]}
            >
                        View Comparison Data
            </h3>
</div>,
          <div
            className="charts"
>
            <ChartArea
                        data={Array []}
            />
</div>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="top-app"
>
                <h1
                                className="title"
                >
                                ADCELLERANT
                </h1>
                <h2
                                className="desc"
                >
                                Digital Advertising & Technology Company
                </h2>
</div>,
              <div
                className="bottom-app"
>
                <h3
                                className="data-desc"
                                onClick={[Function]}
                >
                                View Mobile Data
                </h3>
                <img
                                className="logo"
                                src="creative.svg"
                />
                <h3
                                className="data-desc"
                                onClick={[Function]}
                >
                                View Tablet Data
                </h3>
</div>,
              <h3
                className="data-desc"
                onClick={[Function]}
>
                View Comparison Data
</h3>,
            ],
            "className": "welcome-screen",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="title"
>
                    ADCELLERANT
</h1>,
                  <h2
                    className="desc"
>
                    Digital Advertising & Technology Company
</h2>,
                ],
                "className": "top-app",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ADCELLERANT",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": "ADCELLERANT",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Digital Advertising & Technology Company",
                    "className": "desc",
                  },
                  "ref": null,
                  "rendered": "Digital Advertising & Technology Company",
                  "type": "h2",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="data-desc"
                    onClick={[Function]}
>
                    View Mobile Data
</h3>,
                  <img
                    className="logo"
                    src="creative.svg"
/>,
                  <h3
                    className="data-desc"
                    onClick={[Function]}
>
                    View Tablet Data
</h3>,
                ],
                "className": "bottom-app",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "View Mobile Data",
                    "className": "data-desc",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "View Mobile Data",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "logo",
                    "src": "creative.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "View Tablet Data",
                    "className": "data-desc",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "View Tablet Data",
                  "type": "h3",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "View Comparison Data",
                "className": "data-desc",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "View Comparison Data",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ChartArea
              data={Array []}
/>,
            "className": "charts",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "data": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
